### PR TITLE
Issue - #2809967: Modules and their associated plugins should provide plugin.yml 

### PR DESCRIPTION
**Pull request by scotthooker**
Add plugin_type.yml to commerce_checkout similar to how address does.

### CODE REVIEW

File: modules/checkout/commerce_checkout.plugin_type.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+commerce_checkout.checkout_pane:
+  label: Commerce checkout pane
+  provider: commerce_checkout
+  plugin_manager_service_id: plugin.manager.commerce_checkout_pane
+  plugin_configuration_schema_id: commerce_checkout.commerce_checkout_pane.[id]
+  plugin_definition_decorator_class: Drupal\commerce_checkout\PluginDefinition\CheckoutPanePluginDefinitionDecorator
```
Comment (mglaman) [inline modules/checkout/commerce_checkout.plugin_type.yml]: We don't have this class.

File: modules/checkout/commerce_checkout.plugin_type.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+commerce_checkout.checkout_flow:
+  label: Commerce checkout flow
+  provider: commerce_checkout
+  plugin_manager_service_id: plugin.manager.commerce_checkout_flow
+  plugin_configuration_schema_id: commerce_checkout.commerce_checkout_flow.[id]
+  plugin_definition_decorator_class: Drupal\commerce_checkout\PluginDefinition\CheckoutFlowPluginDefinitionDecorator
```
Comment (mglaman) [inline modules/checkout/commerce_checkout.plugin_type.yml]: This also does not exist.

**Review by mglaman — CHANGES_REQUESTED** [review]

**Comment by scotthooker** [comment]
As there is nothing special about our plugins we can just provide the default one from plugin. 
What other commerce sub modules contain plugins? 
1. Checkout
   - checkout flow
   - checkout pane
     ...

File: modules/checkout/commerce_checkout.plugin_type.yml (hunk 2 above)
Comment (scotthooker) [reply modules/checkout/commerce_checkout.plugin_type.yml]: As there is nothing special about our plugins we can just provide the default one from plugin. Updated.

File: modules/checkout/commerce_checkout.plugin_type.yml (hunk 1 above)
Comment (scotthooker) [reply modules/checkout/commerce_checkout.plugin_type.yml]: As there is nothing special about our plugins we can just provide the default one from plugin. Updated

**Comment by mglaman** [comment]
No idea. As far as I know this is just for the Plugin module, and not 100% sure what this really gains us. Just some developer experience around plugins? Or what does it provide as a direct gain.

**Comment by scotthooker** [comment]
We supply one in address module - as far as the developer experience goes I use it like I do the old views_field list in d7. 

I included it in my own custom module prior to porting it to here once seeing we include it in address.

**Comment by scotthooker** [comment]
@bojanz any specific reason that you added the definition to address module?

**Comment by bojanz** [comment]
It's integration for the Plugin contrib, Bart wanted it, I figured it doesn't hurt.

Our other plugin types live in commerce_payment and commerce_promotion.

**Comment by scotthooker** [comment]
I'll add the yaml for them too. I think its quite handy.

**Comment by scotthooker** [comment]
Added definitions for payment and promotion

**Comment by bojanz** [comment]
Def doesn't hurt. Added trailing newlines and some spaces, committed in 7c327cb0288e100e6d01de45f01c549604563c72.

**Pull request closed without merging**